VisualizationBase/src/Scene.cpp

```diff
@@ -175,24 +177,48 @@ bool Scene::switchToView(const QString viewName)
 	return view != nullptr;
 }
 
-void Scene::addViewItem(ViewItem *view)
+void Scene::addViewItem(ViewItem *view, QPoint position)
 {
 	Q_ASSERT(!inAnUpdate_);
-	viewItems_.append(view);
+	if (position.x() <= 0 || position.y() <= 0)
+		position = nextEmptyPosition();
+	if (viewItems_[position.x()].size() <= position.y())
+		viewItems_[position.x()].resize(position.y() + 1);
+	//If there already is an item -> use insert
+	if (viewItems_[position.x()][position.y()])
```

Comment:
If the rest of the code works fine even if we had a 3x5 grid, then you don't need to add checks. However I'm sure if the viewSwitcher will work as expected if the grid is bigger. Do you have an idea?